comp/core/flare/types/types.go

```diff
@@ -9,147 +9,27 @@
 package types
 
 import (
+	flaredef "github.com/DataDog/datadog-agent/comp/core/flare/def"
 	"go.uber.org/fx"
 )
 
 // FlareBuilder contains all the helpers to add files to a flare archive.
-//
-// When adding data to a flare the builder will do multiple things internally.
-//
-// First a log of the archive creation will be kept and shipped with the flare. When using the the FlareBuilder you
-// should not stop at the first error. We want to collect as much as possible. Any error returned by the a FlareBuilder
-// method is added to the flare log. In general, you can safely ignore those errors unless sending a flare without some
-// file/information would not make sense.
-//
-// The builder will automatically scrub any sensitive data from content and copied files. Still carefully analyze what
-// is being added to ensure that it contains no credentials or unnecessary user-specific data. The FlareBuilder scrubs
-// secrets that match pre-programmed patterns, but it is always better to not capture data containing secrets, than to
-// scrub that data.
-//
-// Everytime a file is copied to the flare the original permissions and ownership of the file is recorded (Unix only).
-//
-// There are reserved path in the flare: "permissions.log" and "flare-creationg.log" (both at the root of the flare).
-// Note as well that the flare does nothing to prevent files to be overwritten by different calls. It's up to the caller
-// to make sure the path used in the flare doesn't clash with other modules.
-type FlareBuilder interface {
-	// IsLocal returns true when the flare is created by the CLI instead of the running Agent process. This happens
-	// when the CLI could not reach the Agent process to request a new flare. In that case a flare is still created
-	// directly from the CLI process and will not contains any runtime informations.
-	IsLocal() bool
+// see the aliased type for the full description
+type FlareBuilder = flaredef.FlareBuilder
```

Comment:
Do we need the type alias? Could we update the places where we use `flare/types` and use `flare/def` ?

Reply:
We could, I just thought this was easier for now, to keep the PR delta small.

Reply:
Seems the proliferation of aliases is increasing. I would recommend making the find and replace on this PR and remove the use of alias 😄

Reply:
There is another issue though. I am pretty sure components using fx (the old style) will need to keep using the `flare/types` versions, while components using multi-impl (the new style) will need to use `flare/def` versions. This is because fx-aware types need to use `fx.Out` and regular types need to use `compdef.Out`; the error checking I added in https://github.com/DataDog/datadog-agent/pull/25679 should catch misuses. With this alias in place, this rule is easy to follow, but if we remove the alias, existing components will need to import both `flare/types` and `flare/def` and use both in the same file.